runtime/JavaScript/src/antlr4/InputStream.js

```diff
@@ -6,21 +6,43 @@
 //
 
 var Token = require('./Token').Token;
+require('./polyfills/codepointat');
 
 // Vacuum all input from a string and then treat it like a buffer.
 
 function _loadString(stream) {
 	stream._index = 0;
 	stream.data = [];
-	for (var i = 0; i < stream.strdata.length; i++) {
-		stream.data.push(stream.strdata.charCodeAt(i));
+	var decodeDataAt;
+	if (stream.decodeToUnicodeCodePoints) {
+		decodeDataAt = function(i) {
+			var codePoint = stream.strdata.codePointAt(i);
+			stream.data.push(codePoint);
+			return codePoint <= 0xFFFF ? 1 : 2;
+		};
+	} else {
+		decodeDataAt = function(i) {
+			var codeUnit = stream.strdata.charCodeAt(i);
+			stream.data.push(codeUnit);
+			return 1;
+		};
+        }
+	for (var i = 0; i < stream.strdata.length; ) {
+		i += decodeDataAt(i);
```

Comment:
given the simplicity of the 2 functions, and they are not reused, it might be faster to inline them